lib/data/datasources/category/category_remote_datasource.dart

```diff
@@ -53,6 +52,8 @@ class CategoryRemoteDataSourceImpl implements CategoryRemoteDataSource {
       'type': category.type.name,
       'name': category.name,
       'description': category.description,
+      'icon': category.icon?.content,
+      'icon_type': category.icon?.type.name,
```

Comment:
This is gotten via a relationship on the backend, and we don't store this on the model tables, this is why I recommen, the frontend/client does it like tha too.

In the future we should have tighter testing to make sure the models on both sides have exactly th esame properties.